src/engine.rs

```diff
@@ -151,16 +151,19 @@ fn handle_update(
             info!("Received BlockValid or BlockInvalid message; ignoring");
         }
         Ok(Update::BlockCommit(block_id)) => node.on_block_commit(block_id, state)?,
-        Ok(Update::PeerMessage(message, sender_id)) => {
-            // Since the signer ID is verified by the validator, we can use it to ensure that this
-            // message was generated by the sender
+        Ok(Update::PeerMessage(message, _)) => {
+            // Since the signer ID in the PeerMessageHeader is verified by the validator, it can be
+            // ensured that the PbftMessage was in fact created and signed by the node that it
+            // claims to be from by comparing the header's signer and the PbftMessage's signer
+            let verified_signer_id = message.header.signer_id.clone();
```

Comment:
You don't need to clone this, since you can compare it as a reference.

Reply:
The message is moved into `ParsedMessage::from_peer_message` though

Reply:
Yeah, that makes sense.